packages/contracts-bridge/contracts/BridgeRouter.sol

```diff
@@ -225,6 +244,72 @@ contract BridgeRouter is Version0, Router {
         IBridgeToken(_currentRepr).mint(msg.sender, _bal);
     }
 
+    // ============ Internal: Send ============
+
+    /**
+     * @notice Debit tokens from the function caller
+     * as part of sending tokens across chains
+     * @dev Locks canonical tokens in escrow in BridgeRouter
+     * OR Burns representation tokens
+     * @param _token The token to pull from the sender
+     * @param _amount The amount to pull from the sender
+     * @return _tokenId the bytes canonical token identifier
+     * @return _detailsHash the hash of the canonical token details (name, symbol, decimal)
+     */
+    function _debitTokens(address _token, uint256 _amount)
+        internal
+        returns (bytes29 _tokenId, bytes32 _detailsHash)
+    {
+        // ensure that amount is non-zero
+        require(_amount > 0, "!amnt");
+        // Setup vars used in both if branches
+        IBridgeToken _t = IBridgeToken(_token);
+        // remove tokens from circulation on this chain
+        if (tokenRegistry.isLocalOrigin(_token)) {
+            // if the token originates on this chain,
+            // hold the tokens in escrow in the Router
+            IERC20(_token).safeTransferFrom(msg.sender, address(this), _amount);
+            // query token contract for details and calculate detailsHash
+            _detailsHash = BridgeMessage.getDetailsHash(
+                _t.name(),
+                _t.symbol(),
+                _t.decimals()
+            );
+        } else {
+            // if the token originates on a remote chain,
+            // burn the representation tokens on this chain
+            _t.burn(msg.sender, _amount);
+            _detailsHash = _t.detailsHash();
+        }
+        // get the tokenID
+        (uint32 _domain, bytes32 _id) = tokenRegistry.getTokenId(_token);
+        _tokenId = BridgeMessage.formatTokenId(_domain, _id);
+    }
+
+    /**
+     * @notice Dispatch a message via Nomad to a destination domain
+     * addressed to the remote BridgeRouter on that chain
+     * @dev Message will trigger `handle` method on the remote BridgeRouter
+     * when it is received on the destination chain
+     * @param _destination The domain of the destination chain
+     * @param _tokenId The canonical token identifier for the transfer message
+     * @param _action The contents of the transfer message
+     */
+    function _sendTransferMessage(
```

Comment:
non-functional change should be in a separate PR

Reply:
https://github.com/nomad-xyz/monorepo/pull/385